FormSubmissionAndValidation2/src/components/CheckBox.jsx

```diff
@@ -0,0 +1,33 @@
+import PropTypes from "prop-types";
+import { ErrorMessage } from "./ErrorMessage";
+import { userFormConfig } from "../helper/userHelper";
+import React from "react";
+
+export const CheckBox = React.memo(function CheckBox({fieldId, value, errorMessage, updateValue}) {
+  const {label, message} = userFormConfig[fieldId];
+  
+  const checked = value ? "checked" : ""
```

Comment:
Is this required?

Reply:
until the user explicitly checks the box the value would come as undefined. but here the value would be either true or false. So may be `const checked = value || false;` make more sense.

Reply:
Do we need a comparison at all? What would be the default value here if a user has not interacted with the component. Will it be truthy or falsy?